src/jit/lsra.cpp

```diff
@@ -2641,6 +2644,26 @@ regMaskTP LinearScan::getKillSetForNode(GenTree* tree)
             }
             break;
 
+#ifdef _TARGET_X86_
+        case GT_INTRINSIC:
+            if (compiler->compFloatingPointUsed)
```

Comment:
This is the same check performed above at line 2603: https://github.com/dotnet/coreclr/pull/7023/files#diff-93fb9832a03a9783e2a415b581fcb45cL2603. I don't think it's necessary, so I'm happy to remove it.